### ISSUE TITLE
Datadog provider doesn't seems to be working

### ISSUE THREAD

**Issue by hari2192** [argoproj/argo-rollouts#817]
Datadog provider doesn't seems to be working for us.

I have updated argo-rollouts with latest version v0.9.2.

Also, followed instructions from here https://argoproj.github.io/argo-rollouts/features/analysis/#datadog-metrics

1. Created secret for datadog API key and APP key in argo-rollouts namespace:

```
apiVersion: v1
kind: Secret
metadata:
  name: datadog
type: Opaque
data:
  address: https://api.datadoghq.com
  api-key: <datadog-api-key>
  app-key: <datadog-app-key>
```

 2. Created analysis in application namespace:
```
apiVersion: argoproj.io/v1alpha1
kind: AnalysisTemplate
metadata:
  name: pod-running
spec:
  metrics:
  - name: pod-running
    interval: 5m
    successCondition: result >= 0
    failureLimit: 3
    provider:
      datadog:
        interval: 5m
        query: |
          sum:kubernetes.pods.running{env:dev,account:sandbox}
```

3. Ran the analysis but getting this error by describing analysis run:
```
Status:
  Message:  analysis spec invalid: metrics[0]: no provider specified
  Phase:    Error
```

Is there anything I am missing here ? @stephen-harris @jessesuen 

_Originally posted by @hari2192 in https://github.com/argoproj/argo-rollouts/issues/702#issuecomment-721535027_

**Comment by jessesuen** [comment]
DataDog is not available in v0.9. Its coming in v0.10

**Comment by jessesuen** [comment]
FYI if you'd like to try it earlier than v0.10 (available in ~3 weeks), you can install https://github.com/argoproj/argo-rollouts/blob/master/manifests/install.yaml. 

I will update docs to indicate the feature is available since v0.10

Duplicate of https://github.com/argoproj/argo-rollouts/issues/702
